homeassistant/components/esphome/__init__.py

```diff
@@ -222,6 +222,7 @@ async def send_home_assistant_state_event(event: Event) -> None:
             # Only communicate changes to the state or attribute tracked
             if (
                 "old_state" in event.data
+                and event.data["old_state"] is not None
```

Comment:
Use `event.data.get("old_state") is not None` to combine the check on this line and on the previous line

Reply:
Done.